src/utils/helpers.ts

```diff
@@ -0,0 +1,93 @@
+import {IActivity, ITesterDetails, ITestResult} from "../models";
+import {uniq, max, maxBy} from "lodash";
+import dateFns from "date-fns";
+import {TIMES} from "./Enums";
+
+export const getStaleOpenVisits = (openVisits: IActivity[]): IActivity[] => {
+  const warningTime = dateFns.subHours(new Date(), TIMES.NOTIFICATION_TIME);
+  return openVisits
+    .filter((a) => a.activityType === "visit")
```

Comment:
You could have added used the `activityType` filter provided by the activities service

Reply:
I could, and investigated that, but I also need to handle the case that there's a more recent wait activity on the books for them which would put them outside the action windows. It's possibly slightly excessive, but safer than not.